client/notifications/socket-handlers.ts

```diff
@@ -25,13 +25,29 @@ const eventToAction: EventToActionMap = {
     }
   },
 
+  clear: event => {
+    const { timestamp } = event
+    return {
+      type: '@notifications/clearBegin',
```

Comment:
This is just a memory leak if used like this. You either need to:

- Add a separate action for this
- Dispatch both a `clearBegin` and a `clear`
or
- Make the `clear` case in the reducer attempt to remove items as well (if you do the grouping by request ID thing I suggested, you could do this only in the case that you didn't have a request ID or it wasn't in the map)

Reply:
What about leaving this code here as it is, and instead have the `clearBegin` action in the reducer not attempt to cache any values if the request ID is not sent?

Reply:
I think semantically that's a bit weird, since elsewhere in the app, `fooBegin` means `foo` is about to happen, and `foo` is the "commit" of that action. I'd prefer to keep those semantics here as well.

I get that you're trying to reuse code in the reducer, but I think consistent semantics are a better thing to optimize for here